assets/app/scripts/directives/events.js

```diff
@@ -5,22 +5,106 @@ angular.module('openshiftConsole')
       restrict: 'E',
       scope: {
-        resourceKind: "@",
-        resourceName: "@",
+        resourceKind: "@?",
+        resourceName: "@?",
         projectContext: "="
       },
       templateUrl: 'views/directives/events.html',
       controller: function($scope){
+        $scope.events = {};
+        $scope.filter = {
+          text: ''
+        };
+
+        var filterForResource = function(events) {
+          if (!$scope.resourceKind || !$scope.resourceName) {
+            return events;
+          }
+
+          return _.filter(events, function(event) {
+            return event.involvedObject.kind === $scope.resourceKind &&
+                   event.involvedObject.name === $scope.resourceName;
+          });
+        };
+
+        var sortedEvents = [];
+        var sortEvents = function() {
+          // TODO: currentField is renamed in angular-patternfly 3.0
+          var sortID = _.get($scope, 'sortConfig.currentField.id', 'lastTimestamp'),
+              order = $scope.sortConfig.isAscending ? 'asc' : 'desc';
+          sortedEvents = _.sortByOrder($scope.events, [sortID], [order]);
+        };
+
+        var applyFilter = $filter('filter');
+        var filterForKeyword = function() {
+          $scope.filteredEvents = applyFilter(sortedEvents, $scope.filter.text);
+        };
+
+        $scope.$watch('filter.text', _.debounce(function() {
+          $scope.$apply(filterForKeyword);
+        }), 250, { maxWait: 500 });
 
-        var filterEvent = function(event) {
-          return (event.involvedObject.kind === $scope.resourceKind) && (event.involvedObject.name === $scope.resourceName);
+        var update = function() {
+          // Sort first so we can update the filter as the user types without resorting.
+          sortEvents();
+          filterForKeyword();
         };
 
+        // Invoke update when first called, debouncing subsequent calls.
+        var debounceUpdate = _.debounce(function() {
+          $scope.$evalAsync(update);
+        }, 250, {
+          leading: true,
+          trailing: false,
+          maxWait: 1000
+        });
+
+        // Set up the sort configuration for `pf-simple-sort`.
+        $scope.sortConfig = {
+          fields: [{
+            id: 'lastTimestamp',
+            title: 'Time',
+            sortType: 'alpha'
+          }, {
+            id: 'type',
+            title: 'Severity',
+            sortType: 'alpha'
+          }, {
+            id: 'reason',
+            title: 'Reason',
+            sortType: 'alpha'
+          }, {
+            id: 'message',
+            title: 'Message',
+            sortType: 'alpha'
+          }, {
+            id: 'count',
+            title: 'Count',
+            sortType: 'numeric'
+          }],
+          isAscending: false,
+          onSortChange: update
+        };
+
+        // Conditionally add kind and name to sort fields if not passed to the directive.
+        if (!$scope.resourceKind || !$scope.resourceName) {
```

Comment:
That's correct. The `pf-simple-sort` widget doesn't support changing config currently, although I think it's been fixed in Patternfly 3.1.0.